config/default.schema.json

```diff
@@ -418,7 +423,25 @@
 		"BLOCK_DISPOSABLE_EMAIL_DOMAINS",
 		"HOST",
 		"ACTIVATION_LINK_PERIOD_OF_VALIDITY_SECONDS",
-		"LERNSTORE_MODE",
-		"PROMETHEUS"
-	]
+		"LERNSTORE_MODE"
+	],
+	"allOf": [
+		{
+			"$ref": "#/definitions/FEATURE_PROMETHEUS_ENABLED"
+		}
+	],
+	"definitions": {
+		"FEATURE_PROMETHEUS_ENABLED": {
+			"if": {
+				"properties": {
+					"FEATURE_PROMETHEUS_ENABLED": {
+						"const": true
+					}
+				}
+			},
+			"then": {
+				"required": ["PROMETHEUS"]
+			}
+		}
+	}
```

Comment:
what is the difference to the old one?

Reply:
here, PROMETHEUS is required only if the feature is enabled, otherwise the settings can be skipped.